src/mme-app/actionHandlers/attachActionHandlers.cpp

```diff
@@ -38,11 +54,99 @@
 #include <utils/mmeCauseUtils.h>
 #include "gtpCauseTypes.h"
 
+
+
+#define MCC_MNC_LEN 4
+#define LB_HB_LEN 8
+
 using namespace SM;
 using namespace mme;
 using namespace cmn;
 using namespace cmn::utils;
 
+cmn::ipc::IpcAddress destAddr_dns;
+struct CS_Q_msg cs_msg;
+extern mme_config g_mme_cfg;
+extern MmeIpcInterface* mmeIpcIf_g;
+Void SetDNSConfiguration()
+{
+	DNS::Cache::setRefreshConcurrent( g_mme_cfg.dns_config.concurrent);
+	DNS::Cache::setRefreshPercent( g_mme_cfg.dns_config.percentage);
+	DNS::Cache::setRefreshInterval( g_mme_cfg.dns_config.interval_seconds );
+	DNS::Cache::getInstance().addNamedServer(g_mme_cfg.dns_config.dns1_ip);
+	DNS::Cache::getInstance().applyNamedServers();
+
+}
+
+	template <class T>
+std::string hexFormatWithoutCommas(T value, int width = sizeof(T)*2)
+{
+	struct Numpunct : public std::numpunct<char>
+	{
+		protected:
+			virtual char do_thousands_sep() const { return ' '; }
+			virtual std::string do_grouping() const { return ""; }
+	};
+	std::stringstream ss;
+	ss.imbue({std::locale(), new Numpunct});
+	ss << std::setfill('0') << std::setw(width) << std::hex << value;
+	return ss.str();
+}
+
+
+
+void Ipv4_gateway_ip( EPCDNS::StringVector &ipv4_ip, unsigned long * sgw_ip)
+{
+	EPCDNS::StringVector::const_iterator it = ipv4_ip.begin();
+	std::cout<<*it <<"\n";
+	char * result = (char*)ipv4_ip[0].c_str();
+	std::cout << result << "\n";
+	unsigned long tmp = htonl(inet_addr(result));
+	memcpy(sgw_ip,&tmp, sizeof(unsigned long));
+	std::cout << *sgw_ip << "\n";
+}
+void Extract_IPs( EPCDNS::NodeSelectorResultList &at, unsigned long  *sgw_ip)
+{	
+	EPCDNS::NodeSelectorResultList::const_iterator it = at.begin();
+	Ipv4_gateway_ip((*it)->getIPv4Hosts(), sgw_ip);
+}
+
+void process_dns_req(void *node_obj, unsigned long *sgw_ip)
+{
+	EPCDNS::NodeSelector *ns = static_cast<EPCDNS::NodeSelector *>(node_obj);
+	ns->process();
+	ns->dump();
+	Extract_IPs(ns->getResults(),sgw_ip);
+}
+
+#if 0
+extern "C" Void NodeSelector_test_callback(EPCDNS::NodeSelector &ns, cpVoid data)
+{
+	unsigned long sgw_ip;
+	std::cout << "*********** Asynchronous Node Selector ***********" << std::endl;
```

Comment:
Use log_msg instead of std::cout.

Reply:
removed the C like calls and replaced the std::cout with log_msg